### ISSUE TITLE
amzi_apls_win64 lacks atcltk.lsx

### ISSUE THREAD

**Issue by csrabak** [AmziLS/distribution#10]
Present versions of 64 bit Amzi! Prolog don't have in binary distro the module atcltk.lsx.

I surmise it is not by design because the 32 bit version has it.

**Comment by DennisMerritt** [comment]
To be honest, I don’t really remember but I think there was some issue with the 64-bit Tcl/Tk libraries.  The sources are all up on GitHub if you would like to try to build it.  If you get it working we can update the distribution.

—Dennis

**Comment by csrabak** [comment]
Thanks for repply.

In order to build the lsx do I need a compatible C compiler (like MSVC) which supports both Tcl/Tk **and** Amzi! Prolog?

I ask because I attemting to gage the size of task before jumping in. . .

--
Cesar Rabak

**Comment by DennisMerritt** [comment]
Yes.  The size of the task depends on how much C++ programming you’ve done.  The builds are all set on Windows for Visual Studio.

Two people that I know of have rebuilt all the sources, so I know it’s not too difficult if you understand and like that sort of thing.

Let me know how it goes,
—Dennis

**Comment by csrabak** [comment]
I'll check which versions I've to download to have Tcl/Tk and Amzi! happy. 

Let's see if the free tools Microsoft releases are par to the task.

Will report. 

Thanks for the inputs.

**Comment by RodneyALudwig** [comment]
Are you looking to get the GUI tools working? or the engine for prolog 
programming?

I have the engine working as a library on windows, IOS and android, and 
I am actively
programming some prolog projects.  Windows done with Visual Basic 2008 
(9.0),
IOS uses XCode, and android uses Android Studio.

I do not use any tools except those for the command line, as I wrote my 
own debugger.

Let me know.
Rod Ludwig


------ Original Message ------

**Comment by csrabak** [comment]
I'm looking to get the Tcl/Tk connection to work with Amzi! Prolog.

In order this to work an extension on the Amzi! side has to be present (atcltk.lsx).

In fact, in the `samples/tcltk/lb_query` a demo of the project is available.

So my search is to have this solution for GUI related programming without having to deal with the platform's specific API for showing menus, etc.

At least in principle, this approach would be multiplatform...

**Comment by DennisMerritt** [comment]
That is exactly what the purpose of the Tcl/Tk LSX was!  I hope you get it working on 64-bits.  I wonder if there have been updates since?

—Dennis

**Comment by RodneyALudwig** [comment]
Not sure where you are going with this, but I have some experience with tk/TCL before switching to wxpython in a python environment and now some experience in ios and android. For my two cents, I would leave my prolog code in a library, and for multiplatform access use flutter. tk/tcl won't get you too far in the mobile world. Not trying to start a gui cross platform flaming war, just a one time injection and I won't bother you two further with such thoughts. Remember, apple will only allow static libraries.


Sent from my Verizon, Samsung Galaxy smartphone

-------- Original message --------

**Comment by csrabak** [comment]
I'm not contemplating for the type of work I envisage using Amzi! (of for the sake of this discussion any other Prolog incarnation) any mobile platform, at least for now.

As this has been explained I'm closing the "issue" and if any good result comes from my experiments I'll get back to see how to contribute.

**Comment by RodneyALudwig** [comment]
Got it.
Do keep me in the loop for your experiments.
Best Regards,
Rod Ludwig

------ Original Message ------
